### DISCUSSION TITLE
[SV Retry] Failed validation of retry message

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#373] [Q&A] [answered]
### Which section?

Fast Data

### Which is the version?

SVC v6.4.0

### Description

Hi!

I set up the Single View Creator plugin with the [Retry Mechanism](https://docs.mia-platform.eu/docs/fast_data/configuration/single_view_creator/#single-view-retry).

However, when I try to update a projection changes to test an error with the following MongoDB command:

```js
db.pcCollection.updateOne({'identifier.id': 1}, {$set: {changes: [{state: "NEW"}] }})
```

The following retry message is generated:

```json
{
  "singleViewName": "test-single-view",
  "singleViewIdentifier": {
    "id": 1
  },
  "change": {},
  "__internal__kafkaInfo": {},
  "retry": {
    "lastError": {
      "type": "UNKNOWN_ERROR",
      "message": "error"
    },
    "attempts": 0
  }
}
```

However, when the retry message is later processed by the SVC, I receive the following error:

```json
{"level":40,"time":170533345364,"hostname":"retry-svc","key":"{\"id\":1 }","cause":"message received did not pass validation","errors":"[{\"instancePath\":\"/value/__internal__kafkaInfo\",\"schemaPath\":\"#/properties/value/properties/__internal__kafkaInfo/required\",\"keyword\":\"required\",\"params\":{\"missingProperty\":\"topic\"},\"message\":\"must have required property 'topic'\"}]","msg":"failed to process message and aggregate corresponding single view"}
```

The retry message is then discarded.

Is this the intended behavior? If yes, what is the proper way to force a projection change with the retry mechanism?

**Comment by ianFar96** [comment, marked as answer]
As we discussed in private, the problem comes from the Projection Changes message that didn't have the proper format, thus the SVC generated a bad sv-retry message. We'll implement a more strict validation on the Projection Change record so the validation message appears upon consuming the record.

The problem fixes itself when providing the correct fields inside the changes array object in the Projection Changes record as defined here https://docs.mia-platform.eu/docs/fast_data/inputs_and_outputs#projection-changes

**Reply by flower-of-the-bridges** [reply]
Hi @ianFar96 thank you very much!
